rest/route/agent.go

```diff
@@ -1546,3 +1544,104 @@ func (h *checkRunHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(fmt.Sprintf("Successfully created check run for  '%v'", t.Id))
 }
+
+// GET /rest/v2/task/{task_id}/github_dynamic_access_token/{owner}/{repo}
+type createGitHubDynamicAccessToken struct {
+	owner  string
+	repo   string
+	taskID string
+
+	permissions github.InstallationPermissions
+
+	env evergreen.Environment
+}
+
+func makeCreateGitHubDynamicAccessToken(env evergreen.Environment) gimlet.RouteHandler {
+	return &createGitHubDynamicAccessToken{
+		env: env,
+	}
+}
+
+func (g *createGitHubDynamicAccessToken) Factory() gimlet.RouteHandler {
+	return makeCreateGitHubDynamicAccessToken(g.env)
+}
+
+func (g *createGitHubDynamicAccessToken) Parse(ctx context.Context, r *http.Request) error {
+	if g.owner = gimlet.GetVars(r)["owner"]; g.owner == "" {
+		return errors.New("missing owner")
+	}
+	if g.repo = gimlet.GetVars(r)["repo"]; g.repo == "" {
+		return errors.New("missing repo")
+	}
+	if g.taskID = gimlet.GetVars(r)["taskID"]; g.taskID == "" {
+		return errors.New("missing taskID")
+	}
+
+	err := utility.ReadJSON(r.Body, &g.permissions)
+
+	errorMessage := fmt.Sprintf("reading permissions body for task '%s'", g.taskID)
+	grip.Error(message.WrapError(err, message.Fields{
+		"message": errorMessage,
+		"task_id": g.taskID,
+	}))
+
+	return errors.Wrapf(err, errorMessage)
+}
+
+func (g *createGitHubDynamicAccessToken) Run(ctx context.Context) gimlet.Responder {
+	t, err := task.FindOneId(g.taskID)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrapf(err, "finding task '%s'", g.taskID))
+	}
+	if t == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", g.taskID),
+		})
+	}
+
+	p, err := model.FindMergedProjectRef(t.Project, t.Version, true)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(err)
+	}
+	if p == nil {
+		return gimlet.MakeJSONErrorResponder(gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("project ref '%s' not found", t.Project),
+		})
+	}
```

Comment:
Can we please add a comment here explaining that the permission we used needs to take into account both the permissions specified in the project settings for the requester and the permissions specified by the command?